generate_network.py

```diff
@@ -4,79 +4,68 @@
 import networkx as nx
 import random, math
 import json
-import argparse, os, sys
+import sys, os
+import string
+import typer
+from enum import Enum
+
+# Consts
+class nw_types(Enum):
+    configmodel = "CONFIGMODEL"
+    scalefree = "SCALEFREE"                     # power law
+    newmanwattsstrogatz = "NEWMANWATTSSTROGATZ" # mesh, smallworld
+    barbell = "BARBELL"                         # partition
+    balancedtree = "BALANCEDTREE"               # committees?
+    star = "STAR"                               # spof
+
+class node_types(Enum):
+    desktop = "DESKTOP"
+    mobile = "MOBILE"
+
+nw_fname = "network_data.json"
+prefix = "waku_"
+
+### I/O related fns ###########################################################
 
 # Dump to a json file
-def write_json(filename, data_2_dump):
-    json.dump(data_2_dump, open(filename,'w'), indent=2)
+def write_json(dirname, json_dump):
+    fname = os.path.join(dirname, nw_fname)
+    json.dump(json_dump, open(fname,'w'), indent=2)
```

Comment:
Use a context manager (`with`) otherwise the file handle keeps open and may leak resources:

```python
...
with open(fname, "w") as f:
    json.dump(json, f, ident=2)
```